### ISSUE TITLE
javascript click not clicking object when ElementNotInteractableException exception thrown

### ISSUE THREAD

**Issue by tvtester** [zebrunner/carina#1571]
**Describe the bug**
JavascriptExecutor to perform click in the doClick() method of ExtendedWebElement class, not performing click on the object. This gets executed when exception ElementNotInteractableException happened while clicking on the element.

```
public void doClick() {
                try {
                    DriverListener.setMessages(Messager.ELEMENT_CLICKED.getMessage(new String[]{ExtendedWebElement.this.getName()}), Messager.ELEMENT_NOT_CLICKED.getMessage(new String[]{ExtendedWebElement.this.getNameWithLocator()}));

                    try {
                        ExtendedWebElement.this.element.click();
                    } catch (ElementNotInteractableException var3) {
                        ExtendedWebElement.LOGGER.warn("Trying to do click by JavascriptExecutor because element '" + ExtendedWebElement.this.getNameWithLocator() + "' is not visible...");
                        JavascriptExecutor executor = (JavascriptExecutor)ExtendedWebElement.this.getDriver();
                        executor.executeScript("arguments[0].click();", new Object[]{ExtendedWebElement.this.element});
                    }
                } catch (WebDriverException var4) {
                    if (var4 == null || !var4.getMessage().contains("Other element would receive the click:")) {
                        throw var4;
                    }

                    ExtendedWebElement.LOGGER.warn("Trying to do click by Actions due to the: " + var4.getMessage());
                    Actions actions = new Actions(ExtendedWebElement.this.getDriver());
                    actions.moveToElement(ExtendedWebElement.this.element).click().perform();
                }

            }
```
In this case, Action methods works fine (it does perform click), but above method is not using Actions class to do. so script is failing as object is not clicked by JavascriptExecutor.

**Logs/Links**
2022-01-06 10:22:49 ExtendedWebElement [TestNG-method=Confirm_continue_watching_is_working_as_expected-1-28] [WARN] Trying to do click by JavascriptExecutor because element 'poster (By.xpath: .//img[@class='thumbnail' and @width])' is not visible...

**Web (please complete the following information):**
 - Browser [e.g. chrome]
 - Version [e.g. 96]

**Additional context**
 - Carina Version [e.g. 7.2.14]

**Comment by vdelendik** [comment]
in `7.3.15-SNAPSHOT` we implemented absolutely different approach having 3 different ways to make a click. Actual decision which one to use should be done based on real application state. So we have now different variants of:
* click
* clickByJs
* clickByActions

Changes are available in `7.3.15-SNAPSHOT` core build

**Comment by vdelendik** [comment]
verified by carina dream team
